protocols/index.md

```diff
@@ -0,0 +1,33 @@
+# CloudState Documentation and Specification format rules
+
+The CloudState Protocol Specification MUST use [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt) definitions.
+
+Use the following formatting and Markdown syntax to define documentation and specification of the CloudState Protocol.
+The template/example below is proven to work well with the `protoc-gen-doc` plugin, so please stick to it 100%.
+
+```
+//The *EntityDiscovery* Service is the entrypoint for CloudState Proxies
+//
+//---
+//
+//`Specification`:
+//
+//  - A `User Function` MUST implement EntityDiscovery
+//  - `discover`
+//    - *MUST* be invoked with a *valid* `ProxyInfo`
+//    - *MUST* be *idempotent*
+//    - *MAY* be called any number of times by its CloudState Proxy
+//    - *MUST* yield a failure if the provided `ProxyInfo` is not deemed compatible with the User Function
```

Comment:
As we start [trying out versioning](https://github.com/cloudstateio/cloudstate/pull/377#issuecomment-693035298) we have to define here how to "yield a failure". I think we can't by protocol?